apps/src/applab/AppLabView.jsx

```diff
@@ -206,7 +206,7 @@ var AppLabView = React.createClass({
 module.exports = connect(function propsFromStore(state) {
   return {
     isReadOnlyWorkspace: state.level.isReadOnlyWorkspace,
-    instructionsInTopPane: state.level.instructionsInTopPane,
+    instructionsInTopPane: state.level.instructionsInTopPane && !!state.level.instructionsMarkdown,
```

Comment:
Ah, agreed.  Defining a property `C` that is derived from `A` and `B` seems cleaner than redefining `A` in terms of `A` and `B`.